terraform/nextstrain_batch.tf

```diff
@@ -0,0 +1,179 @@
+variable "nextstrain-batch-params" {
+  type  = map
+  default = {
+    retry_strategy = 1
+    timeout = 14400
+  }
+}
+
+resource "aws_batch_job_definition" "nextstrain-job-definition" {
+  name = "nextstrain-job-definition"
+  type = "container"
+  parameters = var.nextstrain-batch-params
+
+  container_properties = <<CONTAINER_PROPERTIES
+{
+    "command": [],
+    "image": "nextstrain/base:latest",
+    "memory": 1024,
+    "vcpus": 1,
+    "jobRoleArn": "${aws_iam_role.nextstrain-jobs-role.arn}"
+}
+CONTAINER_PROPERTIES
+}
+
+# resources below needed for compute environment
+
+resource "aws_iam_role" "nextstrain-ecs-instance-role" {
+  name = "nextstrain_ecs_instance_role"
+  assume_role_policy = templatefile("${path.module}/iam_templates/trust_policy.json", {
+    trust_services = ["ec2"]
+  })
+}
+
+
+resource "aws_iam_role_policy_attachment" "nextstrain-ecs-instance-role" {
+  role       = aws_iam_role.nextstrain-ecs-instance-role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonEC2ContainerServiceforEC2Role"
+}
+
+
+resource "aws_iam_instance_profile" "nextstrain-ecs-instance-role" {
+  name = "nextstrain_ecs_instance_profile"
+  role = aws_iam_role.nextstrain-ecs-instance-role.name
+}
+
+
+resource "aws_iam_role" "nextstrain-batch-service-role" {
+  name = "nextstrain_aws_batch_service_role"
+  assume_role_policy = templatefile("${path.module}/iam_templates/trust_policy.json", {
+    trust_services = ["batch"]
+  })
+}
+
+
+resource "aws_iam_role_policy_attachment" "nextstrain-batch-service-role" {
+  role       = aws_iam_role.nextstrain-batch-service-role.name
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSBatchServiceRole"
+}
+
+
```

Comment:
why are these iam roles not in `nextstrain_iam.tf`?

Reply:
i'll move them